dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,75 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+void ParseHeaderMetadata(NumpyParseTarget& target, std::string header) {
+  header.erase(std::remove_if(header.begin(), header.end(), ::isspace), header.end());
+
+  const char t1[] = "{'descr':\'";
+  size_t l1 = strlen(t1);
+  DALI_ENFORCE(strncmp(header.c_str(), t1, l1) == 0);
```

Comment:
I think we can rise some error message like, "Malformed header", maybe we don't need to report how it's malformed but to let user (and us) know what happened in all of the DALI_ENFORCEs in this file.